src/overlays/actors/ovl_En_Clear_Tag/z_en_clear_tag.c

```diff
@@ -61,25 +76,1001 @@ static ColliderCylinderInit D_809D5C40 = {
     },
     { 15, 30, 10, { 0, 0, 0 } },
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Clear_Tag/func_809D35B0.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Clear_Tag/func_809D36A8.s")
+static u32 D_809D5C98 = 0; // unused
+static u32 D_809D5C9C = 0; // unused
+
+static EnClearTagEffect sClearTagEffects[CLEAR_TAG_EFFECT_MAX_COUNT];
+
+#include "../../../../assets/overlays/ovl_En_Clear_Tag/ovl_En_Clear_Tag.c"
+
+/**
+ * Creates a debris effect.
+ * Debris effects are spawned when the Arwing dies. It spawns fire effects.
+ */
+void EnClearTag_CreateDebrisEffect(GlobalContext* globalCtx, Vec3f* position, Vec3f* velocity, Vec3f* acceleration,
+                                   f32 scale, f32 floorHeight) {
+    s16 i;
+    s16 seed;
+    EnClearTagEffect* effect;
+    GlobalContext* globalCtx2;
+
+    globalCtx2 = globalCtx;
+    effect = (EnClearTagEffect*)globalCtx2->specialEffects;
+
+    // Look for an available effect to allocate a Debris effect to.
+    for (i = 0; i < CLEAR_TAG_EFFECT_MAX_COUNT; i++, effect++) {
+        if (effect->type == CLEAR_TAG_EFFECT_AVAILABLE) {
+            effect->type = CLEAR_TAG_EFFECT_DEBRIS;
+
+            effect->position = *position;
+            effect->velocity = *velocity;
+            effect->acceleration = *acceleration;
+
+            effect->scale = scale;
+
+            // Set the debris effects to spawn in a circle.
+            effect->rotationY = Rand_ZeroFloat(M_PI * 2);
+            effect->rotationX = Rand_ZeroFloat(M_PI * 2);
+
+            effect->timer = effect->bounces = 0;
+
+            effect->floorHeight = floorHeight;
+
+            seed = (s32)Rand_ZeroFloat(10.0f);
+            effect->seed = seed;
+
```

Comment:
idk if seed is the right word here. thats usually used as part of the random number generation process.
since its a general purpose random number id prob just call it `rand`

Reply:
I like this suggestion a lot.